redis.go

```diff
@@ -0,0 +1,72 @@
+package iprepd
+
+import (
+	"math/rand"
+	"time"
+
+	"github.com/go-redis/redis"
+	log "github.com/sirupsen/logrus"
+)
+
+type redisLink struct {
+	master      *redis.Client
+	readClients []*redis.Client
+}
+
+func (r *redisLink) del(k ...string) *redis.IntCmd {
+	return r.master.Del(k...)
+}
+
+func (r *redisLink) flushAll() *redis.StatusCmd {
+	return r.master.FlushAll()
+}
+
+func (r *redisLink) get(k string) (ret []byte, err error) {
+	readRot := make([]*redis.Client, len(r.readClients))
+	p := rand.Perm(len(r.readClients))
+	for a, b := range p {
+		readRot[b] = r.readClients[a]
+	}
+	for i := range readRot {
+		ret, err = readRot[i].Get(k).Bytes()
```

Comment:
This will try each read replica in succession as we get failures. That impacts our time bound for end of line response in the worst case. Is it preferable to fail fast here or downstream? Would it make sense to fail to the master if a single replica fails?

Reply:
Yeah, I could go either way here. The intent with this is to maintain the standard operating method, assuming something like a single node failure (probably most likely case). Failing fast means loss of a replica impacts the service which isn't ideal. Redirecting to master on the loss of a single node could do the same.

I'm cool either way; if you think operationally we should change this to behave differently let's do it.

Reply:
I think falling back to successive replicas is fine. Another option would be to make simultaneous requests to multiple replicas (if applicable) and take the first one that we get back. That multiplies load, but we could scale redis and make requests to X of the replicas.

This is all in the interests of optimizing for speed, which it sounded like was the most important factor for the service.

/cc @autrilla